adapters/coap-vertx-base/src/main/java/org/eclipse/hono/adapter/coap/DefaultDeviceResolver.java

```diff
@@ -160,13 +189,18 @@ public SecretKey getKey(final PskPublicInformation identity) {
                 }
             });
         });
+        SecretKey key;
         try {
             // credentials client will wait limited time only
-            return secret.join();
+            key = secret.join();
+            span.setTag("secret", true);
```

Comment:
are you sure that this should be set as a tag instead of just being logged to the span?
if this indeed needs to be a tag then I'd rather have a constant for the Tag or at least its name ...

Reply:
I have no glue, what the difference of a tag or logged should be :-).

Reply:
A tag on a Span is a searchable *label*, if you will. You can use it to find Spans that have a particular tag value. A Span's log is not searchable per se and is usually only shown, once a Span of interest has been identified. From what I see in the code, the outcome of retrieving the secret should either be logged to the Span (`span.log()`) or should be reflected by means of the standard `http.status_code` tag, e.g. setting it to 200 or 404 ...

Reply:
OK, I will adapt it to "log".